core/unit_test/TestDeviceAndThreads.py

```diff
@@ -30,7 +31,22 @@ def GetFlag(flag, *extra_args):
     return int(p.stdout)
 
 def GetNumThreads(max_threads):
-    for x in [1, 2, 3, 5, 7]:
+    args = []
+    name = platform.system()
+    if name == 'Darwin':
+        args = ['sysctl', '-n', 'hw.physicalcpu_max']
+    elif name == 'Linux':
+        args = ['nproc', '--all']
+    else:
+        args = ['wmic', 'cpu', 'get', 'NumberOfCores']
```

Comment:
Why is `os.cpu_count()` no sufficient?
https://docs.python.org/3/library/os.html#os.cpu_count

Reply:
That gives us the logical count, while above I am using the physical count. I figured it was better to rely on the physical core count.